extensions/ql-vscode/src/CodeLensProvider.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  CodeLensProvider,
+  TextDocument,
+  CodeLens,
+  Command,
+  Range
+} from 'vscode';
+
+class QuickEvalCodeLensProvider implements CodeLensProvider {
+  // Each provider requires a provideCodeLenses function which will give the various documents
+  // the code lenses
+  async provideCodeLenses(document: TextDocument): Promise<CodeLens[]> {
+
+    const codeLenses: CodeLens[] = [];
+
+    for (let index = 0; index < document.lineCount; index++) {
+      const textLine = document.lineAt(index);
+      // Match a predicate signature, including predicate name, parameter list, and opening brace.
+      const regex = new RegExp(/\w+\(\s*.*(?:,\s*)*\)\s*\{/);
```

Comment:
It would be _really_ nice to get the name of the predicate in the codelens text. You can do that by capturing the word as below:

```suggestion
      const regex = new RegExp(/(\w+)\(\s*.*(?:,\s*)*\)\s*\{/);
```

Then you can call: `const matches = textLine.text.match(regex);`

`matches[1]` contains the matched word.

Reply:
Made these changes. I also added an additional whitespace match between the word match and opening '(' to account for any predicate definition that has that space.